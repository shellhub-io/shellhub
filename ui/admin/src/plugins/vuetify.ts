// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
