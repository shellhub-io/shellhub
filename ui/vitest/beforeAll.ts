import MockAdapter from "axios-mock-adapter";
import { beforeAll } from "vitest";
import ResizeObserver from "resize-observer-polyfill";
import { systemApi } from "../src/api/http";

const mockedSystemInfo = {
  authentication: {
    local: false,
    saml: true,
  },
  endpoints: {
    api: "localhost:80",
    ssh: "localhost:22",
  },
  setup: true,
  version: "v0.18.3",
};

const systemApiMock = new MockAdapter(systemApi.getAxios());
systemApiMock.onGet("http://localhost:3000/info").reply(200, mockedSystemInfo);

global.CSS = { supports: () => false } as never;
global.ResizeObserver = ResizeObserver;

beforeAll(() => {
  global.CSS = {
    supports: () => false,
    escape: (str: string) => str,
  } as never;
});
