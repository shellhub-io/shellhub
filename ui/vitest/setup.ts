/**
 * Vitest setup function
 */
export async function setup() {
  global.CSS = {
    supports: () => false,
    escape: (str: string) => str,
  } as never;
}

// FAIL LOUDLY on unhandled promise rejections / errors
process.on("unhandledRejection", (reason) => {
  // eslint-disable-next-line no-console
  console.log("FAILED TO HANDLE PROMISE REJECTION");
  throw reason;
});

/**
 * Vitest Teardown function
 */
export async function teardown() {
  console.log("üìù vitest globalTeardown");
}
