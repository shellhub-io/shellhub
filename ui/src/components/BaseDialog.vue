<template>
  <v-dialog
    v-model="showDialog"
    :fullscreen
    :max-width
  >
    <slot />
  </v-dialog>
</template>

<script setup lang="ts">
import { computed } from "vue";
import { useDisplay } from "vuetify";

const props = defineProps<{
  /* Breakpoints
  * - sm: 600px (default)
  * - md: 960px
  * - lg: 1280px
  * - xl: 1920px
  * - xxl: 2560px
  */
  breakpoint?: "sm" | "md" | "lg" | "xl" | "xxl"
  forceFullscreen?: boolean
}>();

const showDialog = defineModel<boolean>({ required: true });
const { smAndDown, thresholds } = useDisplay();
const fullscreen = computed(() => props.forceFullscreen || smAndDown.value);
const maxWidth = computed(() => fullscreen.value ? undefined : thresholds.value[props.breakpoint || "sm"]);

defineExpose({ fullscreen, maxWidth });
</script>
