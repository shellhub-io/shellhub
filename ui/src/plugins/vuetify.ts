// Styles
import "@mdi/font/css/materialdesignicons.css";
import "@fortawesome/fontawesome-free/css/all.css";
import "vuetify/styles";
import { VFileUpload, VFileUploadItem } from "vuetify/labs/VFileUpload";
import "../styles/_variables.scss";
import { aliases as faAliases, fa } from "vuetify/iconsets/fa";

// Vuetify
import { createVuetify } from "vuetify";

const light = {
  dark: false,
  colors: {
    primary: "#667acc",
    secondary: "#FFFFFF",
    background: "#F5F5F5",
    tabs: "#F5F5F5",
    foreground: "#F5F5F5",
    paymentForm: "#F5F5F5",
    terminal: "#0f1526",
    "v-theme-background": "#FFFFFF",
    "v-theme-surface": "#FFFFFF",
    "v-theme-card": "#FFFFFF",
  },
};

const dark = {
  dark: true,
  colors: {
    primary: "#667acc",
    secondary: "#1E2127",
    background: "#18191B",
    tabs: "#1E1E1E",
    foreground: "#1E1E1E",
    paymentForm: "#E0E0E0",
    terminal: "#0f1526",
    "v-theme-background": "#1E2127",
    "v-theme-surface": "#1E2127",
    "v-theme-card": "#22252B",
  },
};

export default createVuetify({
  components: {
    VFileUpload,
    VFileUploadItem,
  },
  theme: {
    defaultTheme: "dark",
    themes: {
      dark,
      light,
    },
  },
  defaults: {
    global: {

    },
    VTextField: {
      variant: "outlined",
      density: "comfortable",
    },
    VSelect: {
      variant: "outlined",
      density: "comfortable",
    },
    VTextarea: {
      variant: "outlined",
    },
  },
  icons: {
    faAliases,
    sets: {
      fa,
    },
  },
});
