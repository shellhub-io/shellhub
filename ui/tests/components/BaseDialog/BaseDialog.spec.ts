import { createVuetify } from "vuetify";
import { DOMWrapper, mount, VueWrapper } from "@vue/test-utils";
import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
import BaseDialog from "@/components/BaseDialog.vue";

vi.mock("vuetify", async () => {
  const actual = await vi.importActual<typeof import("vuetify")>("vuetify");
  return {
    ...actual,
    useDisplay: () => ({
      smAndDown: { value: false },
      thresholds: {
        value: {
          sm: 600,
          md: 960,
          lg: 1280,
          xl: 1920,
          xxl: 2560,
        },
      },
    }),
  };
});

type BaseDialogWrapper = VueWrapper<InstanceType<typeof BaseDialog>>;

describe("BaseDialog", () => {
  const vuetify = createVuetify();
  let dialog: DOMWrapper<HTMLElement>;
  let wrapper: BaseDialogWrapper;

  const breakpointTests = [
    { breakpoint: "sm", expectedWidth: 600 },
    { breakpoint: "md", expectedWidth: 960 },
    { breakpoint: "lg", expectedWidth: 1280 },
    { breakpoint: "xl", expectedWidth: 1920 },
    { breakpoint: "xxl", expectedWidth: 2560 },
  ] as const;

  const slotContent = "<div>Test content</div>";

  const mountWrapper = (props: { forceFullscreen?: boolean; breakpoint?: "sm" | "md" | "lg" | "xl" | "xxl" } = {}) => mount(BaseDialog, {
    global: { plugins: [vuetify] },
    slots: { default: slotContent },
    props: { modelValue: true, ...props },
  });

  beforeEach(async () => {
    wrapper = mountWrapper();
    dialog = new DOMWrapper(document.body);
  });

  afterEach(() => {
    vi.clearAllMocks();
    wrapper.unmount();
  });

  it("Is a Vue instance", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("Renders the component", () => {
    expect(dialog.html()).toMatchSnapshot();
  });

  it("Renders slot content", () => {
    expect(dialog.html()).toContain("Test content");
  });

  it("Uses default sm breakpoint (600px) when no breakpoint is specified", () => {
    expect(wrapper.vm.maxWidth).toBe(600);
    expect(wrapper.vm.fullscreen).toBe(false);
  });

  it("Uses fullscreen mode when forceFullscreen is true", () => {
    wrapper = mountWrapper({ forceFullscreen: true });
    expect(wrapper.vm.maxWidth).toBeUndefined();
    expect(wrapper.vm.fullscreen).toBe(true);
  });

  breakpointTests.forEach(({ breakpoint, expectedWidth }) => {
    it(`Uses correct width for ${breakpoint} breakpoint (${expectedWidth}px)`, () => {
      wrapper = mountWrapper({ breakpoint });

      expect(wrapper.vm.maxWidth).toBe(expectedWidth);
      expect(wrapper.vm.fullscreen).toBe(false);
    });
  });
});
