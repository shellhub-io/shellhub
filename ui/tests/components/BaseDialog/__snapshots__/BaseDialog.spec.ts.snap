// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseDialog > Renders the component 1`] = `
"<body>
  <div class="v-overlay-container">
    <div class="v-overlay v-overlay--active v-theme--light v-locale--is-ltr v-dialog" style="z-index: 2400;" aria-modal="true" role="dialog">
      <transition-stub name="fade-transition" appear="true" persisted="false" css="true">
        <div class="v-overlay__scrim"></div>
      </transition-stub>
      <transition-stub name="dialog-transition" appear="true" persisted="true" css="true">
        <div class="v-overlay__content" style="max-width: 600px;" tabindex="-1">
          <div>Test content</div>
        </div>
      </transition-stub>
    </div>
  </div>
</body>"
`;
